test-data/unit/check-dataclasses.test

```diff
@@ -2489,3 +2489,17 @@ class Base:
 class Child(Base):
     y: int
 [builtins fixtures/dataclasses.pyi]
+
+[case testDunderReplacePresent]
+# flags: --python-version 3.13
+from dataclasses import dataclass
+
+@dataclass
+class Coords:
+    x: int
+    y: int
+
+
+replaced = Coords(2, 4).__replace__(x=2, y=5)
```

Comment:
Test passing only one of `x` and `y`; also test that an error is generated if a positional argument is passed, or if an unrecognized argument is passed.

Reply:
Thanks for added test cases; can you also add something like `Coords(2, 4).__replace__(unknown=5) # E: ...`?

Reply:
Great callout, should have some more cases covered to begin with: https://github.com/python/mypy/pull/17469/commits/e3f2b73e48d106b683a35770eef8d193bd2c815d